### ISSUE TITLE
[Streaming]: No Response:  This camera is not responding

### ISSUE THREAD

**Issue by DitPun** [dgreif/ring#762]
### Streaming Issue

I have a total of 8 Ring cameras of different types, and I'm having a problem opening a video stream for any of them, irrespective of device (macOS 11.5.2, iOS 14.7.1).  However, the snapshots do work okay and update every ten seconds, as set.  Of note, these cameras did work on a previous version of the ring plug in (I'm running Hoobs, FWIW).

Please note the error messages noted below.  

### Proposed Solution

None.

### Environment

HOOBS

### What operating system are you on?

Mac

### Relevant log output

```shell
9/7/2021, 3:52:41 PM [Ring] Preparing Live Stream for Side
9/7/2021, 3:52:42 PM [Ring] Waiting for stream data from Side (0.97s)
9/7/2021, 3:52:44 PM [HOOBS 9F6A@Setup Endpoints] The write handler for the characteristic 'Setup Endpoints' on the accessory 'Side' was slow to respond!
9/7/2021, 3:52:50 PM [HOOBS 9F6A@Setup Endpoints] The write handler for the characteristic 'Setup Endpoints' on the accessory 'Side' didn't respond at all!. Please check that you properly call the callback!
9/7/2021, 3:52:50 PM [Ring] Stopped Live Stream for Side
9/7/2021, 3:52:50 PM [Ring] Failed to prepare stream for Side (9.299s)
9/7/2021, 3:52:50 PM [Ring] Error: 
    at _super (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/createErrorClass.ts:13:22)
    at new EmptyErrorImpl (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/EmptyError.ts:26:3)
    at Object.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/lastValueFrom.ts:72:18)
    at Object.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:194:14)
    at SafeSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at SafeSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at /home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subject.ts:90:30
    at Object.errorContext (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/errorContext.ts:29:5) {
  stack: 'Error: \n' +
    '    at _super (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/createErrorClass.ts:13:22)\n' +
    '    at new EmptyErrorImpl (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/EmptyError.ts:26:3)\n' +
    '    at Object.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/lastValueFrom.ts:72:18)\n' +
    '    at Object.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:194:14)\n' +
    '    at SafeSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)\n' +
    '    at SafeSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)\n' +
    '    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)\n' +
    '    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)\n' +
    '    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)\n' +
    '    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)\n' +
    '    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)\n' +
    '    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)\n' +
    '    at OperatorSubscriber.Subscriber._complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:132:24)\n' +
    '    at OperatorSubscriber.Subscriber.complete (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subscriber.ts:106:12)\n' +
    '    at /home/hoobs/.hoobs/node_modules/rxjs/src/internal/Subject.ts:90:30\n' +
    '    at Object.errorContext (/home/hoobs/.hoobs/node_modules/rxjs/src/internal/util/errorContext.ts:29:5)',
  name: 'EmptyError',
  message: 'no elements in sequence'
```

### Honesty Time

- [X] I swear this is worth your time!

**Comment by DitPun** [comment]
BTW, I've tried uninstalling/reinstalling,  including uninstalling from command line.  I deleted the hoobs hub from Home, all to no avail.

**Comment by dgreif** [comment]
Did you downgrade to the previous version of the plugin that worked?  Other than that, all of my suggestions are in https://github.com/dgreif/ring/wiki/Camera-Troubleshooting

**Comment by superlarry** [comment]
I think I'm having a similar issue (log entries below) since two versions ago.  My other Ring camera (on a floodlight/motion sensor) continues to work fine - it's just the doorbell that doesn't work.  The troubleshooting steps didn't help for me.  Can you point me to how to install an older version so I can help pinpoint the issue?

```
[9/8/2021, 4:00:59 PM] [Ring] Preparing Live Stream for Front Door
[9/8/2021, 4:01:00 PM] [Ring] Waiting for stream data from Front Door (1.063s)
[9/8/2021, 4:01:08 PM] [Ring] Stopped Live Stream for Front Door
[9/8/2021, 4:01:08 PM] [Ring] Failed to prepare stream for Front Door (9.116s)
[9/8/2021, 4:01:08 PM] [Ring] Error: 
    at _super (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/util/createErrorClass.ts:13:22)
    at new EmptyErrorImpl (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/util/EmptyError.ts:26:3)
    at Object.complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/lastValueFrom.ts:72:18)
    at Object.complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:194:14)
    at SafeSubscriber.Subscriber._complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at SafeSubscriber.Subscriber.complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/usr/local/lib/node_modules/homebridge-ring/node_modules/rxjs/src/internal/Subscriber.ts:106:12) {
  message: 'no elements in sequence'
}
```

I'm new to this and eager to help out however I can.

**Comment by superlarry** [comment]
Does the same camera work from inside your Ring app?  Mine didn't, and I ended up having to restart the device - then everything was fine.  Mine is a doorbell, so I had to open it and remove the battery to restart it.

**Comment by dgreif** [comment]
Any luck?

**Comment by pquattro** [comment]
got exactly the same since 2 weeks.

**Comment by superlarry** [comment]
Eventually, mine stopped working both in the Ring app and in Homebridge.  I ended up reconnecting my Ring to my wifi network, and it started working again in both.  I hope it keeps working.

**Comment by dgreif** [comment]
I was able to track down this issue and should have a fix out this weekend.  Sorry for the broken streams!

**Comment by superlarry** [comment]
Thanks for tracking down the error!  Now the exception is handled, but it still doesn't work.  Is there anything we can do to help debug what's causing it?

```
[10/2/2021, 9:23:03 AM] [Ring] Preparing Live Stream for Front Door
[10/2/2021, 9:23:04 AM] [Ring] Waiting for stream data from Front Door (1.128s)
[10/2/2021, 9:23:12 AM] [Ring] Stopped Live Stream for Front Door
[10/2/2021, 9:23:13 AM] [Ring] Stream was closed before it was ready for Front Door (9.902s)
```

**Comment by dgreif** [comment]
@superlarry if your camera isn't working from the Ring app, then the issue is with your camera and not with the plugin.  I suggest you get it working in the Ring app first (may need to contact Ring support), and then it should hopefully work from HomeKit again.

**Comment by pquattro** [comment]
Exactly the same like Superlarry, again. Nobody on cam, i have snapshot every 5 minutes.. nothing else. It work on app, simple app too... strange...

**Comment by x5nder** [comment]
Same issue here. Camera works perfectly in Ring app. No idea which version update broke the snapshots/streaming :-(

```
[11/8/2021, 9:36:30 PM] [Ring] Preparing Live Stream for Front Door
[11/8/2021, 9:36:32 PM] [Ring] Waiting for stream data from Front Door (1.132s)
[11/8/2021, 9:36:32 PM] [Ring] Received stream data from Front Door (1.8s)
[11/8/2021, 9:36:32 PM] [Ring] Stopped Live Stream for Front Door
```

**Comment by dgreif** [comment]
Do any of you have end-to-end encryption enabled?

**Comment by x5nder** [comment]
No (I actually have the 1st gen Ring video doorbell, that doesn't even support E2EE). Weirdly enough, after not having been able to get any new screenshots or camera feed for almost two weeks, I updated Node.js / NPM yesterday and restarted Homebridge... and I just saw that I'm getting new screenshots again. Not sure if these updates are related to the plugin problem or if it's just coincidence, but I'll keep an eye on it.
